src/Symfony/Component/HttpKernel/EventListener/AbstractTestSessionListener.php

```diff
@@ -71,6 +71,13 @@ public function onKernelResponse(FilterResponseEvent $event)
 
         if ($session instanceof Session ? !$session->isEmpty() || (null !== $this->sessionId && $session->getId() !== $this->sessionId) : $wasStarted) {
             $params = session_get_cookie_params();
+
+            foreach ($event->getResponse()->headers->getCookies() as $cookie) {
+                if ($session->getName() === $cookie->getName() && $params['path'] == $cookie->getPath() && $params['domain'] == $cookie->getDomain()) {
```

Comment:
Please use `===` instead of `==`

Reply:
They could be `null` or empty strings.

Reply:
can you add a test case where this would fail when the first `==` is turned into a `===` please?

Reply:
Looking again, the path will always be a string and at least `/`, so can make that comparison strict. The domain, however, is `null` in `Cookie` but an empty string in `$params['domain']` (so the existing test cases already fail if that's made strict).

Reply:
You can always cast the value from `Cookie` to a string and then use strict comparison. I think that would still be better than loose comparison.

Reply:
both sides could need the cast.
but since this is tested, that's not really needed anyway.